lib/Models/REpresentationalStateTransferCatalogFunction.js

```diff
@@ -0,0 +1,540 @@
+var knockout  = require('terriajs-cesium/Source/ThirdParty/knockout');
+var BooleanParameter = require('./BooleanParameter');
+
+/**
+ * A {@link CatalogFunction} that invokes a Representational State Transfer (REST) process.
+ *
+ * @alias REpresentationalStateTransferCatalogFunction
+ * @constructor
+ * @extends CatalogFunction
+ *
+ * @param {Terria} terria The Terria instance.
+ */
+function REpresentationalStateTransferCatalogFunction(terria) {
+    CatalogFunction.call(this, terria);
+
+    /**
+     * Gets or sets the URL of the REST server.  This property is observable.
+     * @type {String}
+     */
+    this.url = undefined;
+
+    /**
+     * Gets or sets the name of this REST process.  This property is observable.
+     * @type {String}
+     */
+    this.name = undefined;
+
+    /**
+     * Gets or sets the description of this REST process.  This property is observable.
+     * @type {String}
+     */
+    this.description = undefined;
+
+    /**
+     * Gets or sets the identifier of this REST process.  This property is observable.
+     * @type {String}
+     */
+    this.identifier = undefined;
+
+    /**
+     * Gets or sets whether to use key value pairs (KVP) embedded in Execute URL, or whether to make a POST request
+     * with XML data.
+     * @type {Boolean}
+     */
+    this.executeWithHttpGet = true;
+
+    /**
+    *Gets or sets the input parameters defined as a set of {@link FunctionParameter} objects. 
+    *@type {Object}
+    */
+    this.input = undefined;
+    knockout.track(this,['_parameters','url','input']);
+    //knockout.track(this, ['_parameters', 'url', 'inputs', 'executeWithHttpGet']);
+}
+
+inherit(CatalogFunction, REpresentationalStateTransferCatalogFunction);
+
+defineProperties(REpresentationalStateTransferCatalogFunction.prototype, {
+    /**
+     * Gets the type of data item represented by this instance.
+     * @memberOf REpresentationalStateTransferCatalogFunction.prototype
+     * @type {String}
+     */
+    type : {
+        get : function() {
+            return 'rest';
+        }
+    },
+
+    /**
+     * Gets a human-readable name for this type of data source, 'Representational State Transfer (REST)'.
+     * @memberOf REpresentationalStateTransferCatalogFunction.prototype
+     * @type {String}
+     */
+    typeName : {
+        get : function() {
+            return 'Representational State Transfer (REST)';
+        }
+    },
+
+    /**
+     * Gets the parameters used to {@link CatalogFunction#invoke} to this process.
+     * @memberOf REpresentationalStateTransferCatalogFunction
+     * @type {CatalogFunctionParameters[]}
+     */
+    parameters : {
+        get : function() {
+            return this._parameters;
+        }
+    },
+
+});
+
+/**
+ * Gets or sets the list of converters between a REST Input and a {@link FunctionParameter}.
+ * @type {Array}
+ */
+REpresentationalStateTransferCatalogFunction.parameterConverters = [
+    {
+        id: 'LiteralData',
```

Comment:
All of this is copy/pasted from WPS, and it's also quite specific to the details of that protocol. Rather than requiring REST inputs to be specified in WPS terms, the `REpresentationalStateTransferCatalogFunction` should expect the user to specify inputs directly using an array of parameter types, e.g. `BooleanParameter` or `PolygonParameter`.